pkg/scheduler/plugins/proportion/capacity_policy/capacity_policy.go

```diff
@@ -83,3 +113,131 @@ func getRequiredQuota(tasksToAllocate []*pod_info.PodInfo) *podgroup_info.JobReq
 	}
 	return &quota
 }
+
+// getFirstPendingPod returns the first pod in a job that is in Pending status.
+// This is used to avoid duplicate quota checks for the same job.
+func getFirstPendingPod(job *podgroup_info.PodGroupInfo) *pod_info.PodInfo {
+	for _, pod := range job.PodInfos {
+		if pod.Status == pod_status.Pending {
+			return pod
+		}
+	}
+	return nil
+}
+
+// OnSessionOpen is called when a new scheduling session begins. It registers
+// the early quota checking function that prevents jobs from being considered
+// for scheduling if they would exceed their parent queues' quotas.
+func (cp *CapacityPolicy) OnSessionOpen(ssn *framework.Session) {
+	// Register early quota checks
+	ssn.AddPrePredicateFn(func(task *pod_info.PodInfo, job *podgroup_info.PodGroupInfo) error {
+		// Only check for the first pending pod to avoid duplicate checks
+		firstPending := getFirstPendingPod(job)
+		if firstPending == nil || task != firstPending {
+			return nil
+		}
+
+		// Check parent queue quotas
+		return cp.checkParentQueueQuotas(job, ssn)
+	})
+}
+
+// checkParentQueueQuotas verifies that a job's resource requirements don't
+// exceed quotas at any level in its queue hierarchy. This includes:
+// - GPU quota checks
+// - CPU quota checks
+// - Memory quota checks
+//
+// The function traverses up the queue hierarchy starting from the job's
+// immediate parent queue. If any quota would be exceeded, it returns an
+// error with a detailed message.
+//
+// Note: Preemptible jobs (PriorityTrainNumber) are allowed to exceed parent
+// queue quotas, while non-preemptible jobs must strictly adhere to quotas.
+func (cp *CapacityPolicy) checkParentQueueQuotas(job *podgroup_info.PodGroupInfo, ssn *framework.Session) error {
+	// Skip quota checks for preemptible jobs
+	if job.Priority == constants.PriorityTrainNumber {
```

Comment:
We aim to separate the job priority and preemptibility soon, and the Train priority is not the only one that is currently preemptible.
We do have several scalars to define the resource of a queue: quota, limit and over quota weight. We should look at limit here and not quota and then it will be correct for all jobs.